frontend/src/contexts/ThemeContext.tsx

```diff
@@ -0,0 +1,50 @@
+import { Dispatch, FC, SetStateAction, createContext, useContext, useEffect, useState } from "react";
+
+interface ThemeContextType {
+    darkTheme: boolean,
+    setDarkTheme: Dispatch<SetStateAction<boolean>>
+}
+
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
+
+export const useTheme = () => {
+    const context = useContext(ThemeContext);
+    if (!context) {
+      throw new Error("useTheme must be used within an ThemeProvider");
+    }
+    return context;
+}
+
+export const ThemeProvider: FC<{children: React.ReactNode}> = ({children}) => {
+    const [darkTheme, setDarkTheme] = useState(() => {
+        if (typeof window !== 'undefined') {
+            const theme = localStorage.getItem('theme');
+          return (theme && theme === 'dark') ? true: false;
+        }
+        return false;
+      });
+
+    useEffect(() => {
```

Comment:
Is it necessary? We are already doing this when initializing `useState`

Reply:
Good call. It used to be necessary but no longer is